### PR TITLE
Re-enable annotations e2e test

### PR DESCRIPTION
**Pull request by kevinwcyu**
- Updates `@grafana/*` packages in order to re-enable the annotations in the e2e test. Packages are updated to v9.3.2 since there were some fixes in the `e2e` and `e2e-selector` packages that are needed for the tests to work.
- Updates `@grafana/aws-sdk` to fix a build error where `className` was not properly recognized as a prop on `ResourceSelector` used in `QueryEditor.tsx`

Note: to run e2e locally, you'll need to make sure you're using and npm version < 9. For more info see https://github.com/grafana/grafana/issues/59425.

### CODE REVIEW

**Review by sunker — APPROVED** [review]
LGTM

File: docker-compose.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
     build:
       context: ./.config
       args:
-        grafana_version: ${GRAFANA_VERSION:-9.1.2}
```
Comment (sarahzinger) [inline docker-compose.yaml:9]: just curious if this change is necessary, seems harmless, more just curious if the e2e tests use this docker file at all

File: src/__mocks__/datasource.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ export const mockDatasource = new DataSource({
   type: 'athena-datasource',
   name: 'Athena Data Source',
   access: 'proxy',
+  readOnly: false,
```
Comment (sarahzinger) [inline src/__mocks__/datasource.ts:40]: what did this do? I'm guessing this is just a new required field in 9.3.2?

File: src/datasource.test.ts [hunk 3]
```diff
@@ -61,61 +56,76 @@ describe('AthenaDatasource', () => {
...
 
   describe('When performing getRegions', () => {
     it('should return a list of regions', async () => {
       const response = setupRegionsResponse();
-      backendSrv.get = jest.fn(() => Promise.resolve(response));
       const regionsResponse = await ctx.ds.getRegions();
 
       expect(regionsResponse).toHaveLength(response.length);
-      expect(regionsResponse).toBe(response);
+      expect(regionsResponse).toEqual(response);
```
Comment (sarahzinger) [inline src/datasource.test.ts:88]: curious if this is the result of just a package update or if this is like meaningfully different in some way?

File: src/datasource.test.ts [hunk 4]
```diff
@@ -61,61 +56,76 @@ describe('AthenaDatasource', () => {
       jsonData: { defaultRegion: 'testRegion', catalog: 'testCatalog', database: 'testDatabase' },
     } as unknown as DataSourceInstanceSettings<AthenaDataSourceOptions>;
     ctx.ds = new DataSource(ctx.instanceSettings);
-    runtime.setBackendSrv(backendSrv as runtime.BackendSrv);
+    ctx.ds.getResource = jest.fn().mockImplementation((path: string) => {
```
Comment (sarahzinger) [inline src/datasource.test.ts:59]: why were these changes needed is setBackendSrv not a thing in the newer version? Looks good tho, just curious!

**Review by sarahzinger — APPROVED** [review]
Looks ok to me! A few questions for my own knowledge, also I think I kind of lost where the resource selector/query editor was involved? Did those changes happen elsewhere or am I just missing it lol.

File: docker-compose.yaml (hunk 1 above)
Comment (kevinwcyu) [reply docker-compose.yaml:9]: the e2e tests that run in ci don't use this docker file, i was using this to test the e2e locally.

File: src/__mocks__/datasource.ts (hunk 2 above)
Comment (kevinwcyu) [reply src/__mocks__/datasource.ts:40]: yea, new required field

File: src/datasource.test.ts (hunk 4 above)
Comment (kevinwcyu) [reply src/datasource.test.ts:59]: `setBackendSrv` is still a thing, but after upgrading, there were errors with needing to mock `fetch` from the `BackendSrv`. i didn't want to spend time mocking out the internals of the `BackendSrv`.

File: src/datasource.test.ts (hunk 3 above)
Comment (kevinwcyu) [reply src/datasource.test.ts:88]: the mock's response was previously set with `backendSrv.get = jest.fn(() => Promise.resolve(response));` (removed line 70), but now it's set up in the beforeEach so now it's not testing for reference equality anymore.
